specification/dartLangSpec.tex

```diff
@@ -19072,23 +19072,27 @@ \section*{Appendix: Integer Implementations}
 IEEE-754 64-bit floating point number with no fractional part.
 This includes positive and negative \Index{infinity},
 which can be reached by overflowing
-(integer division by zero is still not allowed).
+(integer division by zero is still a dynamic error).
 Otherwise valid integer literals (including any leading minus sign)
 that represent invalid JavaScript \code{int} values
-cannot be compiled to JavaScript.
-Operations on integers may lose precision since 64-bit floating point numbers
-are limited to 53 significant bits.
+are compile-time errors when compiling to JavaScript.
+Operations on integers may lose precision,
+because they are represented as 64-bit floating point numbers
+that are limited to 53 significant bits.
 \item[$\bullet$]
 JavaScript \code{int} instances also implement \code{double},
 and integer-valued \code{double} instances also implement \code{int}.
 The \code{int} and \code{double} class are still separate subclasses of the
 class \code{num},
 but \emph{instances} of either class that represent an integer
 act as if they are actually instances of a common subclass implementing both
-\code{int} and \code{double}. Fractional numbers only implement \code{double}.
+\code{int} and \code{double}.
+Fractional numbers only implement \code{double}.
 \item[$\bullet$]
-Bitwise operations on integers (and, or, xor, negate and shifts)
-all truncate the operands to 32-bit values.
+Bitwise operations on integers (and, or, xor, negate, and shifts)
+all truncate the operands to 32-bit values
+and operate on them as unsigned values
```

Comment:
This seems to be where the details go, so:

> Bitwise operations in integers (and, or, xor, negation and shifts)
> all truncate the operands to 32-bit two's complement integers,
> perform 32-bit operations on those,
> and then the resulting 32 bits are then interpreted as a 32-bit unsigned integer.

I don't want to say that the *operands* are unsigned, because `>>` is sign-extending (so it must have a sign, so it can't be "unsigned"). The above description is precisely following the JavaScript spec, which converts bit-operation operands to 32-bit integers using their spec-function ToInt32, then it performs the bit-operation on those bits, and then interpret then as signed or unsigned 32-bit integers. Finally Dart performs an extra `>>>0` on the result, which calls `ToUint32`, to make it unsigned.

Reply:
Done.